### PR TITLE
(BKR-208) Preserve SUT host output

### PR DESCRIPTION
**Pull request by madAndroid**
Adds a method, and a global option, to allow preservation of SUT coloured output

For example:

window_and_1__login_and_comparing_puppetlabs_master___madandroid_bkr-208-preserve-sut-host-output_ _puppetlabs_beaker

### CODE REVIEW

File: lib/beaker/host.rb [hunk 1]
```diff
@@ -234,7 +234,11 @@ def exec command, options={}
         output_callback = nil
       else
         @logger.debug "\n#{log_prefix} #{Time.new.strftime('%H:%M:%S')}$ #{cmdline}"
-        output_callback = logger.method(:host_output)
+        if @options[:preserve_host_output]
```
Comment (anodelman) [inline lib/beaker/host.rb]: Let's call this color_host_output or some such - preserve_host_output makes it sound like you won't get any output without this being set.

**Comment by madAndroid** [comment]
@anodelman - Yes, good point :) I've made the changes suggested

**Comment by anodelman** [comment]
ok to test

**Comment by anodelman** [comment]
Could you add a command line switch for this new option?

**Comment by rick** [comment]
Definitely like this idea. :sparkles:

**Comment by petems** [comment]
:+1:

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by madAndroid** [comment]
Tests pass locally :-/ .. any indication as to why they're failing in CI? difficult to tell with the current lack of visibility

**Comment by anodelman** [comment]
Once I get clear results I'll paste appropriate chunks in a comment.  I'm just getting a bunch of weird output that I don't trust.

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
This, unfortunately, broke beaker-rspec integration:
```
Failed: errored in validate
#<ArgumentError: wrong number of arguments (5 for 4)>
/usr/local/rvm/rubies/ruby-1.9.3-p484/lib/ruby/1.9.1/socket.rb:236
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/net-ssh-2.10.0/lib/net/ssh/transport/session.rb:70
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/net-ssh-2.10.0/lib/net/ssh.rb:221
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/net-ssh-2.10.0/lib/net/ssh.rb:221
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/ssh_connection.rb:45
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/ssh_connection.rb:34
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host.rb:227
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host.rb:259
/usr/local/rvm/rubies/ruby-1.9.3-p484/lib/ruby/1.9.1/benchmark.rb:295
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host.rb:258
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/command_factory.rb:23
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host/unix/pkg.rb:34
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host_prebuilt_steps.rb:123
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host_prebuilt_steps.rb:122
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host_prebuilt_steps.rb:122
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host_prebuilt_steps.rb:98
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/shared/host_manager.rb:104
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/shared/host_manager.rb:93
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/shared/host_manager.rb:92
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/shared/host_manager.rb:92
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/dsl/patterns.rb:32
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/host_prebuilt_steps.rb:91
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/hypervisor.rb:124
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/network_manager.rb:80
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/network_manager.rb:79
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/lib/beaker/network_manager.rb:79
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/lib/beaker-rspec/beaker_shim.rb:42
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/lib/beaker-rspec/spec_helper.rb:47
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core.rb:97
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/lib/beaker-rspec/spec_helper.rb:5
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/lib/beaker-rspec.rb:5
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/lib/beaker-rspec.rb:5
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/lib/beaker-rspec.rb:1
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/spec/spec_helper.rb:3
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/spec/spec_helper.rb:3
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/spec/acceptance/example_spec.rb:1
/var/lib/jenkins/workspace/qe_beaker_intn-sys_vpool-beaker-rspec-support/layout/64mda/platform/debian-7/beaker-rspec/spec/acceptance/example_spec.rb:1
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1327
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1327
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1325
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1325
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:102
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:88
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:73
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:41
/usr/local/rvm/gems/ruby-1.9.3-p484/gems/rspec-core-3.3.2/exe/rspec:4
/usr/local/rvm/gems/ruby-1.9.3-p484/bin/rspec:23
/usr/local/rvm/gems/ruby-1.9.3-p484/bin/rspec:23
/usr/local/rvm/gems/ruby-1.9.3-p484/bin/ruby_executable_hooks:15
/usr/local/rvm/gems/ruby-1.9.3-p484/bin/ruby_executable_hooks:15
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```
Let me look it over and see what went wrong.

**Comment by anodelman** [comment]
Hm, that smoketest may just be busted.

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
Failures unrelated to patch.

**Pull request merged**